Meme/Meme/MemeTemplatesViewModel.swift

```diff
@@ -0,0 +1,41 @@
+//
+//  MemeTemplatesViewModel.swift
+//  Meme
+//
+//  Created by Mesfin Bekele Mekonnen on 7/30/19.
+//  Copyright © 2019 Mesfin Bekele Mekonnen. All rights reserved.
+//
+
+import Foundation
+
+protocol TemplatesViewModel {
+    func requestTemplates(completion: @escaping ([Template]) -> Void?)
+    func getMeme(template: Template) -> Meme?
+}
+
+struct MemeTemplatesViewModel: TemplatesViewModel {
+    private let networkClient: NetworkClient
+
+     init(networkClient: NetworkClient) {
+        self.networkClient = networkClient
+    }
+}
+
+extension MemeTemplatesViewModel {
+    func requestTemplates(completion: @escaping ([Template]) -> Void?) {
```

Comment:
You don’t need the extension here

Reply:
Yeah, I didn't have it as an extension initially, and decided to move it here. What's a general rule of thumb about when to put things in an extension?